pyrdp/mitm/FakeServer.py

```diff
@@ -0,0 +1,274 @@
+#
+# This file is part of the PyRDP project.
+# Copyright (C) 2022
+# Licensed under the GPLv3 or later.
+#
+import os, random, shutil, socket, subprocess, threading, time
+from tkinter import *
+from PIL import Image, ImageTk
+from pyvirtualdisplay import Display
+from pyrdp.logging import log  # TODO: this logs to GLOBAL and not to the session, setup proper logging
+
+BACKGROUND_COLOR = "#044a91"
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+class FakeLoginScreen:
+    def __init__(self, width=1920, height=1080):
+        self.clicked = False
+
+        # root window
+        # right now all Xephyr instances are merged together because of Tk but I don't know why this happens
+        # asked here: https://stackoverflow.com/questions/74552455/
+        self.root = Tk()
+        self.root.attributes("-fullscreen", True)
+        self.root.geometry(f"{width}x{height}")
+        # TODO: only accepts main return key (not from numpad)
+        self.root.bind("<Return>", self.on_click)
+
+        self._set_background(width, height)
+        self._set_entries()
+
+        # frames for loading animation
+        self.frame_count = 50
+        self.frames = [
+            PhotoImage(
+                file=BASE_DIR + "/images/WindowsLoadingScreenSmall.gif",
+                format=f"gif -index {i}",
+                master=self.root,
+            )
+            for i in range(self.frame_count)
+        ]
+
+        # label for loading animation
+        self.label_loading_animation = Label(self.root, borderwidth=0)
+
+    def _set_background(self, width=1920, height=1080):
+        # background file
+        self.background_image = Image.open(
+            BASE_DIR + "/images/WindowsLockScreen.png"
+        ).resize((width, height))
+        self.background = ImageTk.PhotoImage(self.background_image, master=self.root)
+
+        # background label
+        if (
+            hasattr(self, "label_background")
+            and self.label_background is not None
+            and not self.clicked
+        ):
+            self.label_background.destroy()
+        self.label_background = Label(self.root, image=self.background, borderwidth=0)
+        self.label_background.place(x=0, y=0)
+        self.label_background.lower()
+
+    def _set_entries(self):
+        # username entry
+        self.entry_username = Entry(
+            self.root,
+            font=("Segoe UI", 13),
+            bd=2,
+            bg="white",
+            insertofftime=600,
+            insertwidth="1p",
+            highlightthickness=1,
+            highlightbackground="gray",
+            highlightcolor="#eaeaea",
+        )
+        self.entry_username.place(
+            relx=0.5, rely=0.61, anchor=CENTER, height=40, width=290
+        )
+        self.entry_username.focus()
+
+        # password entry
+        self.entry_password = Entry(
+            self.root,
+            show="•",
+            font=("Segoe UI", 20),
```

Comment:
Does this add special dependencies?

Reply:
Good point! I'm not sure honestly, it's certainly not on my Kali VM and I never exactly looked at it. Also I'm not a fonts expert :smile:. I removed it for now, since it doesn't add much.

Reply:
I'm switching to DejaVu Sans but that's not available on Windows I think. I also didn't find a way to only specify a font family (e. g. Sans Serif)..